runtime/commonMain/src/kotlinx/serialization/internal/PluginGeneratedSerialDescriptor.kt

```diff
@@ -120,8 +120,10 @@ internal inline fun <reified SD : SerialDescriptor> SD.equalsImpl(
 internal inline fun SerialDescriptor.hashCodeImpl(typeParams: Array<SerialDescriptor>): Int {
     var result = serialName.hashCode()
     result = 31 * result + typeParams.contentHashCode()
-    val elementDescriptors = elementDescriptors()
-    result = 31 * result + elementDescriptors.map { it.serialName }.hashCode()
-    result = 31 * result + elementDescriptors.map { it.kind }.hashCode()
+    val elementDescriptors = elementDescriptors
+    val namesHash = elementDescriptors.fold(1) { hash, element -> 31 * hash + element.serialName.hashCode() }
```

Comment:
if you're so eager to remove allocation in `map` (which looks like a micro-optimization to me, since the result is cached anyway), you can use `.asSequence`. Or write `.hashByKey(key: (T) -> K): Int` function to hide `fold` as an implementation detail and reduce amount of copy-pasted code

Reply:
`asSequence().hashCode()` invokes identity hashcode.

>Or write .hashByKey(key: (T) -> K): Int

I prefer not to introduce unnecessary abstractions unless there is real need

Reply:
It really improves readability imo